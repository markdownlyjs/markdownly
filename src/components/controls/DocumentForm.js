import React from 'react';
import PropTypes from 'prop-types';

const DocumentForm = ({ handleSubmit, handleChange, inputValue }) => {
  return (
    <form onSubmit={handleSubmit}>
      <input value={inputValue} onChange={handleChange} />
      <button>+</button>
    </form>
  );
};

DocumentForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  inputValue: PropTypes.string.isRequired
};

export default DocumentForm;
