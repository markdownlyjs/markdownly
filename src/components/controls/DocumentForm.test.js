import React from 'react';
import { shallow } from 'enzyme';
import DocumentForm from './DocumentForm';

describe('DocumentForm component', () => {
  it('renders DocumentForm', () => {
    const wrapper = shallow(<DocumentForm 
      inputValue="hi"
      handleSubmit={() => {}} 
      handleChange={() => {}} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
