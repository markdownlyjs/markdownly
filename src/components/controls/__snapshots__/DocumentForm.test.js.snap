// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentForm component renders DocumentForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    value="hi"
  />
  <button>
    +
  </button>
</form>
`;
